extension/source/hooks/useInputDecode.tsx

```diff
@@ -2,100 +2,52 @@ import { ethers } from 'ethers';
-      payable: false,
-      type: 'function',
-    },
-  ];
+const getMethodFromRegistry = async (data: string) => {
+  if (data === '0x') return { funcSig: 'SENDING ETH', args: [] };
 
-  return new ethers.Contract(address, abi, await provider.read());
-};
-
-const getMethodFromOnChainRegistry = async (
-  data: string,
-  provider: IReadableCell<ethers.providers.Provider>,
-) => {
-  if (data === '0x') return 'SENDING ETH';
-
-  const methodID = ethers.utils.hexDataSlice(data, 0, 4);
-  const registry = await getParitySigRegistry(provider);
-
-  return registry.entries(methodID);
-};
+  const funcSigHex = data.slice(0, 10);
 
-const getMethodFromEtherscan = async (to: string, data: string) => {
   const res = await axios.get(
-    `https://api.etherscan.io/api?module=contract&action=getabi&address=${to}`,
+    `https://sig.eth.samczsun.com/api/v1/signatures?all=true&function=${funcSigHex}`,
   );
+  const funcSig = res.data.result.function[funcSigHex][0].name;
+  const iface = new ethers.utils.Interface([`function ${funcSig}`]);
+  const { args } = iface.parseTransaction({ data });
 
-  if (res.data.result !== 'Contract source code not verified') {
-    const iface = new ethers.utils.Interface(res.data.result);
-    return iface.parseTransaction({ data, value: 1 }).name;
-  }
-
-  assert(false, () => new Error('Unverified Contract'));
+  return { funcSig, args };
 };
 
-const formatMethod = (method: string) =>
-  method
-    .split('(')[0]
-    .replace(/([a-z](?=[A-Z]))/g, '$1 ')
-    .toUpperCase();
-
 type UseInputDecodeValues = {
   loading: boolean;
   method: string;
+  args: ethers.utils.Result;
 };
 
-export const useInputDecode = (
-  functionData: string,
-  to: string,
-): UseInputDecodeValues => {
+export const useInputDecode = (functionData: string): UseInputDecodeValues => {
   const quill = useQuill();
 
   const [loading, setLoading] = useState<boolean>(true);
   const [method, setMethod] = useState<string>('CONTRACT INTERACTION');
+  const [allArgs, setAllArgs] = useState<ethers.utils.Result>([]);
 
   useEffect(() => {
     const getMethod = async () => {
       setLoading(true);
 
       const data = functionData?.replace(/\s+/g, '');
-
-      try {
-        const registryPromise = getMethodFromOnChainRegistry(
-          data,
-          quill.ethersProvider,
-        );
-        const etherScanPromise = getMethodFromEtherscan(to, data);
-        const rawMethod = (await registryPromise) ?? (await etherScanPromise);
-        if (rawMethod) {
-          setMethod(formatMethod(rawMethod));
-        }
-      } catch (error) {
-        console.log({ error });
-      }
+      const { funcSig, args } = await getMethodFromRegistry(data);
+      setMethod(funcSig);
+      setAllArgs(args);
 
       setLoading(false);
     };
 
     if (functionData) {
       getMethod();
     }
-  }, [functionData, to, quill]);
+  }, [functionData, quill]);
 
-  return { loading, method };
+  return { loading, method, args: allArgs };
```

Comment:
It might be a good idea to memoize this to prevent unnecessary renders in consuming components.

Reply:
This one's a bit tricky. This hook is called inside TransactionCard component at the top level and the component is only consumed in [Confirm](https://github.com/web3well/bls-wallet/blob/main/extension/source/Confirm/Confirm.tsx#L111) component. 

The way it's setup is that when a multi-action tx is called, it renders a a new TransactionCard with the current tx calldata as "props". When user uses the arrows to move to the next tx the props given to TransactionCard change which triggers a re-render. Re-rendering will always take place because the props change and since the value changes are handled in a custom hook, memoizing the values won't be helpful here - [reference](https://reactjs.org/docs/react-api.html#reactmemo). 

The only way to get some performance benefit is by caching the API call itself. Can go this route if you think it's useful. The current thought is to use session storage for storing the calldata-functionSig pair. Shouldn't be big.

Reply:
Ah ok then yes, we can leave as is. Caching would make sense for more frequent interactions, but is likely overkill at this point. Can do that as separate work if needed.